### PR TITLE
Sprint 4

### PR DESCRIPTION
**Pull request by dimaspirit**
**Made/Proposed changes:**
- New build tools (Gulp instead of Grunt);
- QB.Recorder;
- Stream Managment;
- Rewrote contributing files (contributors make only pull request into master, a member of team make a release);
- new commands: npm run develop, npm run build, npm run start;
- added to 'gitignore' sublime generated files;
- In build added 'jshint' command, fixed all warnings;
- remove unminified version of SDK. Uses source maps for debugging.
- Remove samples 'node_js' as duplicate 'chat_bots'.

**How should this be manually tested?**
- `npm run start` and check all samples.

**Does the documentation need an update?**
- Update contributors section;
- See section 'ADD to DOCS'.

ADD to DOCS 
1. Record (https://quickblox.com/developers/Sample-webrtc-web)
Support: Firefox 47, Chrome 49, Opera 40.
You can record a stream or a few streams by QB.Recorder. 
Now QB.Recorder in beta version, so be careful with this functionality.

```javascript
var options = {
  callbacks: {
    onStartRecording: function startRecord() {  /* code here */  },
    onErrorRecording: function errorRecord(error) { /* code here */ },
    onPauseRecording: function pauseRecord() { /* code here */ },
    onStopRecording: function stopRecord(blob) { /* code here */ },
    onResumeRecording: function resimeRecord() { /* code here */ }
  }
}
var recorder = new QB.Recorder(stream, options);
recorder.record();
```
known issues:
* We recommend that you limit the duration of recording 3 minutes currently. The long duration can lead to the crash of the tab.
* Firefox has an issue when we are recording a remote stream. Got 'GenericError '. We are not recommended to use QB.Recorder in Firefox because he has some unexpected error when MediaStream was ended, it's crash a page.
* Firefox will crash if a remote stream is broken.

2. Sent messages (https://quickblox.com/developers/Web_XMPP_Chat_Sample)
This feature defines an approach for ensuring that message was delivered to server and was saved.
This feature is unenabled by default.

To enable this features add to config 
```javascript
streamManagement: {
  enable: true
}

// Uses listener by QB.chat.onSentMessageCallback
QB.chat.onSentMessageCallback = function (messageLost, messageSent) {
  if (messageLost) {
    console.error('sendErrorCallback', messageLost);
  } else {
    console.info('sendMessageSuccessCallback', messageSent);
  }
};
```

### CODE REVIEW

**Comment by dimaspirit** [comment]
@IegorKozakov , @Vladlukhanin hey guys, i added editorconfig so please [install a plugin](https://github.com/sindresorhus/editorconfig-sublime#readme) if you need

File: src/modules/qbAuth.js [hunk 1]
```diff
@@ -1,9 +1,4 @@
-/*
- * QuickBlox JavaScript SDK
- *
- * Authentication Module
- *
- */
+'use strict';
```
Comment (soulfly) [inline src/modules/qbAuth.js:1]: это не нарушит ничего?

File: samples/recorder.html [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.1.1/jquery.min.js"></script>
+<script src="../quickblox.min.js"></script>
+<script>
+    /* globals QB*/
+    'use strict';
+
+    var CONFIG = {
+        debug: true,
+        webrtc: {
```
Comment (soulfly) [inline samples/recorder.html]: а нужны конфиги вебртс для этого семпла рекордера?

File: src/modules/qbAuth.js (hunk 1 above)
Comment (dimaspirit) [reply src/modules/qbAuth.js:1]: Директива проигнорируется, если исполниться на старых версиях компилятора.

File: samples/recorder.html (hunk 2 above)
Comment (dimaspirit) [reply samples/recorder.html]: Я удалю сэмпл "recorder.html", когда бу вливать. Сейчас это нужно только для тестирования.

File: spec/SpecRunner.html [hunk 3]
```diff
@@ -13,7 +13,7 @@
 
   <!-- include source files here... -->
   <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
-  <script type="text/javascript" src="../quickblox.min.js"></script>
+  <script type="text/javascript" src="../quickblox.js"></script>
```
Comment (Vladlukhanin) [inline spec/SpecRunner.html]: src="../quickblox.js" - точно? не min должно быть?

**Comment by Vladlukhanin** [comment]
+

**Pull request merged**